core/app.go

```diff
@@ -102,6 +103,15 @@ func NewApp(configFlag string) (*App, error) {
 	a.Tasks = cfg.Tasks
 	a.Telemetry = cfg.Telemetry
 	a.ConfigFlag = configFlag
+
+	// set an environment variable for each service IP address so that
+	// forked processes have access to this information
+	for _, service := range a.Services {
+		envKey := fmt.Sprintf("CONTAINERPILOT_%v_IP",
+			strings.ToUpper(service.Name))
```

Comment:
According to the [documentation](https://www.consul.io/docs/agent/services.html) 

> DNS-compliant service and tag names may contain any alpha-numeric characters, as well as dashes. Dots are not supported because Consul internally uses them to delimit service tags.

I think we should adopt the same convention. service names should be alpha-numeric with dashes.  We'll normalize to uppercase and convert dashes to underscores to comply with environment variables.

If the service name does not conform, then it doesn't get environment variables.